test/shared/definition/ApiDefinitionFactorySpec.scala

```diff
@@ -21,99 +21,98 @@ import shared.config.Deprecation.NotDeprecated
-          }
-        }
-      }
+    "the 'apiStatus' parameter is present but invalid" should {
+      s"default to alpha" in new Test {
+        setupMockConfig(Version9)
+        MockAppConfig.apiStatus(Version9) returns "not-a-status"
 
-      "the 'apiStatus' parameter is present and invalid" should {
-          List(Version3, Version4).foreach { version =>
-            s"default to alpha for $version " in new Test {
-              MockAppConfig.apiStatus(version) returns "ALPHO"
-              MockAppConfig
-                .deprecationFor(version)
-                .returns(NotDeprecated.valid)
-                .anyNumberOfTimes()
-              apiDefinitionFactory.checkBuildApiStatus(version) shouldBe ALPHA
-            }
-          }
+        apiDefinitionFactory.checkBuildApiStatus(Version9) shouldBe ALPHA
       }
+    }
+
+    "the 'deprecatedOn' parameter is missing for a deprecated version" should {
+      "throw an exception" in new Test {
+        MockAppConfig.apiStatus(Version9) returns "DEPRECATED"
+
+        MockAppConfig
+          .deprecationFor(Version9)
+          .returns("deprecatedOn date is required for a deprecated version".invalid)
+          .anyNumberOfTimes()
 
-      "the 'deprecatedOn' parameter is missing for a deprecated version" should {
-        "throw exception" in new Test {
-          MockAppConfig.apiStatus(Version3) returns "DEPRECATED"
-          MockAppConfig
-            .deprecationFor(Version3)
-            .returns("deprecatedOn date is required for a deprecated version".invalid)
-            .anyNumberOfTimes()
+        val exception: Exception = intercept[Exception] {
+          apiDefinitionFactory.checkBuildApiStatus(Version9)
+        }
 
-          val exception: Exception = intercept[Exception] {
-            apiDefinitionFactory.checkBuildApiStatus(Version4)
-          }
+        val exceptionMessage: String = exception.getMessage
+        exceptionMessage shouldBe "deprecatedOn date is required for a deprecated version"
+      }
+    }
+  }
 
-          val exceptionMessage: String = exception.getMessage
-          exceptionMessage shouldBe "deprecatedOn date is required for a deprecated version"
+  "confidenceLevel" when {
+    List(
+      (true, ConfidenceLevel.L250, ConfidenceLevel.L250),
+      (true, ConfidenceLevel.L200, ConfidenceLevel.L200),
+      (false, ConfidenceLevel.L200, ConfidenceLevel.L50)
+    ).foreach { case (definitionEnabled, configCL, expectedDefinitionCL) =>
+      s"confidence-level-check.definition.enabled is $definitionEnabled and confidence-level = $configCL" should {
+        s"return confidence level $expectedDefinitionCL" in new Test {
+          MockAppConfig.confidenceLevelCheckEnabled returns ConfidenceLevelConfig(
+            confidenceLevel = configCL,
+            definitionEnabled = definitionEnabled,
+            authValidationEnabled = true)
+          apiDefinitionFactory.confidenceLevel shouldBe expectedDefinitionCL
```

Comment:
Not new, but why is it the `confidenceLevel` method in `ApplicationDefinitionFactory` the thing that munges the confidence level. Feels like it ought to be in the `AppConfig` itself (in the implicit `ConfigLoader`) rather than being split into two places. The AppConfig `confidenceLevel` method is used in a couple of places directly, but not sure whether this is deliberate.

Reply:
could probably go in the same follow-up story